### PR TITLE
Gradle & Spring Boot - API Rest | MiguelAValdesG

### PR DESCRIPTION
**Pull request by MiguelAValdesG**
* HackerRank Challenges of Gradle &
* HackerRank Challenges of Spring Boot - API Rest

By: Miguel a valdes g

### CODE REVIEW

File: MiguelAValdesG/HackerRank/gradle/currencyFormatter/src/main/java/currencyFormatter/JavaCurrencyFormatter.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    Scanner readLineTerminal = new Scanner(System.in);
+    double payment = readLineTerminal.nextDouble();
+    readLineTerminal.close();
+
+    String unitedStatesPayment = NumberFormat.getCurrencyInstance(Locale.US).format(payment);
+    String indiaPayment = NumberFormat.getCurrencyInstance(new Locale("en", "IN")).format(payment);
+    String chinaPayment = NumberFormat.getCurrencyInstance(Locale.CHINA).format(payment);
+    String francePayment = NumberFormat.getCurrencyInstance(Locale.FRANCE).format(payment);
+
+   log.info("US: " + unitedStatesPayment);
```
Comment (globant-bootcamp) [inline MiguelAValdesG/HackerRank/gradle/currencyFormatter/src/main/java/currencyFormatter/JavaCurrencyFormatter.java:23]: Fix Indentation

File: MiguelAValdesG/HackerRank/gradle/intToString/src/main/java/intToString/JavaIntToString.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    try {
+      Scanner readLineTerminal = new Scanner(System.in);
+      int number = readLineTerminal.nextInt();
+      readLineTerminal.close();
+
+      String numberInString = Integer.toString(number);
+
+      if (number == Integer.parseInt(numberInString)) {
+        log.info("Good job");
```
Comment (globant-bootcamp) [inline MiguelAValdesG/HackerRank/gradle/intToString/src/main/java/intToString/JavaIntToString.java:23]: Could be good idea put this hardcoded string inside static final variables

File: MiguelAValdesG/HackerRank/gradle/intToString/src/main/java/intToString/JavaIntToString.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      } else {
+        log.error("Wrong answer.");
```
Comment (globant-bootcamp) [inline MiguelAValdesG/HackerRank/gradle/intToString/src/main/java/intToString/JavaIntToString.java:25]: Could be good idea put this hardcoded string inside static final variables

File: MiguelAValdesG/HackerRank/springBootAPIRest/src/main/java/com/bootcamp/springBootAPIRest/service/impl/AnagramsServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      char[] secondWordArray = secondWord.toLowerCase().toCharArray();
+
+      Arrays.sort(firstWordArray);
+      Arrays.sort(secondWordArray);
+      return Arrays.equals(firstWordArray, secondWordArray);
+    }
+  }
+
+  public String anagrams(String firstWord, String secondWord) {
+    boolean response = isAnagram(firstWord, secondWord);
```
Comment (globant-bootcamp) [inline MiguelAValdesG/HackerRank/springBootAPIRest/src/main/java/com/bootcamp/springBootAPIRest/service/impl/AnagramsServiceImpl.java:26]: Fix Indentation

File: MiguelAValdesG/HackerRank/springBootAPIRest/src/main/java/com/bootcamp/springBootAPIRest/service/impl/PatternSyntaxCheckerServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+@Service
+public class PatternSyntaxCheckerServiceImpl implements PatternSyntaxCheckerService {
+
+  @Override
+  public List<String> patternSyntaxChecker(String[] patterns) {
+    Logger log = Logger.getLogger(Constants.PATTERN_SYNTAX_CHECKER);
+    List<String> response = new ArrayList<>();
```
Comment (globant-bootcamp) [inline MiguelAValdesG/HackerRank/springBootAPIRest/src/main/java/com/bootcamp/springBootAPIRest/service/impl/PatternSyntaxCheckerServiceImpl.java:20]: It is recommended that the name of the collections and maps start with singular name and them the name of the interface, example: numberList

**Review by globant-bootcamp — CHANGES_REQUESTED** [review]

File: MiguelAValdesG/HackerRank/springBootAPIRest/src/main/java/com/bootcamp/springBootAPIRest/service/impl/AnagramsServiceImpl.java (hunk 4 above)
Comment (MiguelAValdesG) [reply MiguelAValdesG/HackerRank/springBootAPIRest/src/main/java/com/bootcamp/springBootAPIRest/service/impl/AnagramsServiceImpl.java:26]: I didn't see any problem with the indentation in this part of code, in fact, I "ran" the automatic reformat in InteliJ (ctrl + alt + l), but everything was ok and didn't have any change.

**Review by globant-bootcamp — APPROVED** [review]

**Pull request merged**